### PR TITLE
Add ADOT to prod bundle

### PR DESCRIPTION
**Pull request by ivyostosh**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: generatebundlefile/data/input_120_prod.yaml [hunk 1]
```diff
@@ -9,6 +9,13 @@ packages:
         registry: public.ecr.aws/eks-anywhere
         versions:
             - name: 0.1.2-latest-helm
+  - org: aws-observability
+    projects:
+      - name: adot
+        repository: adot/charts/aws-otel-collector
+        registry: public.ecr.aws/eks-anywhere
+        versions:
+            - name: 0.21.1-971c31238aeea5f1bda6ec95caaf24993b304157
```
Comment (a-cool-train) [inline generatebundlefile/data/input_120_prod.yaml:18]: Is this required to be hard coded? Would it be possible to just use latest here?

Comment (ivyostosh) [reply generatebundlefile/data/input_120_prod.yaml:18]: For prod, it was intentional to make it hard coded so we don't auto bundle a version.

**Pull request merged**